### PR TITLE
Keep default values in df_kwargs when calling load_data()

### PR DESCRIPTION
**Pull request by ugGit**
This PR implements issue https://github.com/redcap-tools/PyCap/issues/268.

Previously, the default values in the df_kwargs were overridden, even if no value was specified for them. 
Now, the default values remain unchanged (e.g., for `index_cols` when using `export_records(df_kwargs{'low_memory': False})`).
Additionally, three test cases are implemented to verify the above described behavior, as well as the existing one.

Besides, the command to install poetry in the contribution guidelines is update.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ This package uses [`poetry`](https://python-poetry.org/docs/master/#installation
 1. Install `poetry`
 
 ```sh
-$ curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/install-poetry.py | python -
+$ curl -sSL https://install.python-poetry.org | python3 -
```
Comment (pwildenhain) [inline CONTRIBUTING.md:21]: Thanks for updating this!

File: redcap/methods/base.py [hunk 2]
```diff
@@ -353,7 +353,6 @@ def _return_data(
         ],
         format_type: Literal["json", "csv", "xml", "df"],
         df_kwargs: Optional[Dict[str, Any]] = None,
-        record_type: Literal["flat", "eav"] = "flat",
```
Comment (pwildenhain) [inline redcap/methods/base.py:356]: Is there a reason you've removed this argument? I would still like to be able to have the user return a data frame for the `eav` record type option

**Review by pwildenhain — COMMENTED** [review]
PR looks great 🤩 just one consideration I have for the `eav` record type. 

If the user asks for that then they'll be given one row per _column_ per record, instead of the usual one row per record. So in that case we don't want to define an `index_col`

File: redcap/methods/base.py (hunk 2 above)
Comment (ugGit) [reply redcap/methods/base.py:356]: Not sure anymore. I guess, I did not see, that `content = 'report` when calling `_load_data()` from `export_records()`. I modified the code to leave the `df_kwargs` unchanged for the `eav` record type option.

**Review by pwildenhain — APPROVED** [review]
🙇🏻‍♂️ Thanks for contributing this improvement 🎉

**Pull request merged**